src/move/econia/sources/market.move

```diff
@@ -2530,9 +2513,11 @@ module econia::market {
     ) acquires OrderBooks {
         // Assert valid order restriction flag.
         assert!(restriction <= N_RESTRICTIONS, E_INVALID_RESTRICTION);
-        assert!(price != 0, E_PRICE_0); // Assert nonzero price.
+        // assert!(price != 0, E_PRICE_0); // Assert nonzero price.
         // Assert price is not too high.
-        assert!(price <= HI_PRICE, E_PRICE_TOO_HIGH);
+        if (price > HI_PRICE) {
+            price = HI_PRICE;
+        };
```

Comment:
If there is no check against the price being 0, then a user will be able to place a maker order on the book with price 0.

If the concern here is to make sure that market orders can fill adequately, then we could easily define `MIN_PRICE = 1` and have market order functions pass that in.

Also per above it looks like we don't strictly need this and so I'd advocate for minimum refactor, and also the doc comment hasn't been updated to reflect the changes.

@bogberry your thoughts?